src/components/CellSegmentation/CellSegmentation.tsx

```diff
@@ -0,0 +1,263 @@
+import { useFormikContext } from 'formik';
+import React from 'react';
+import { CommentFieldsFragment, LabwareFlaggedFieldsFragment, SlideCosting } from '../../types/sdk';
+import LabwareScanner from '../labwareScanner/LabwareScanner';
+import MutedText from '../MutedText';
+import Heading from '../Heading';
+import { motion } from 'framer-motion';
+import variants from '../../lib/motionVariants';
+import Panel from '../Panel';
+import RemoveButton from '../buttons/RemoveButton';
+import FormikInput from '../forms/Input';
+import { getCurrentDateTime } from '../../types/stan';
+import CustomReactSelect, { OptionType } from '../forms/CustomReactSelect';
+import { selectOptionValues } from '../forms';
+import Table, { TableBody, TableCell, TableHead, TableHeader } from '../Table';
+import WorkNumberSelect from '../WorkNumberSelect';
+import LabwareSamplesTable from './LabwareSamplesTable';
+import { capitalize } from 'lodash';
+
+type CellSegmentationPageProps = {
+  comments: CommentFieldsFragment[];
+  isQc: boolean;
+};
+
+type CellSegmentationProps = {
+  labware: LabwareFlaggedFieldsFragment;
+  costing?: string;
+  workNumber: string;
+  performed: string;
+  comments: string[];
+};
+
+type CellSegmentationFormProps = {
+  cellSegmentation: CellSegmentationProps[];
+  workNumberAll: string;
+  performedAll: string;
+  costingAll?: string;
+  commentsAll: string[];
+};
+
+const slideCostingOptions: OptionType[] = Object.values(SlideCosting).map((val) => {
+  return { value: capitalize(val), label: val };
+});
+export const Segmentation = ({ comments, isQc }: CellSegmentationPageProps) => {
+  const { values, setFieldValue, setValues } = useFormikContext<CellSegmentationFormProps>();
+  return (
+    <motion.div variants={variants.fadeInWithLift} className="space-y-4 mb-6">
+      <Heading level={3}>Labware</Heading>
+      {values.cellSegmentation.length === 0 && <MutedText>Scan a piece of labware to get started</MutedText>}
+      <LabwareScanner
+        onAdd={async (labware) => {
+          await setValues((prev: CellSegmentationFormProps) => {
+            let cellSegmentation = [...prev.cellSegmentation];
+            cellSegmentation.push({
+              labware: labware,
+              workNumber: prev.workNumberAll ?? '',
+              performed: prev.performedAll ?? '',
+              comments: prev.commentsAll ?? [],
+              costing: !isQc ? prev.costingAll ?? '' : undefined
```

Comment:
For `costing` here there are two defaults: `''` and `undefined`. Do they have separate use cases inside this field?

Reply:
yes for QC, costing is not part of the form, so it should be set to undefined, for cell segmentation form, costing should be init to an empty string (as it is mandatory field there)